src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -40,6 +47,14 @@ public function __construct(LoggerInterface $logger = null)
     }
 
     /**
+     * @internal
+     */
+    public function setArgumentResolverManager(ArgumentResolverManager $argumentResolverManager)
```

Comment:
Its here purely for backwards compability (hence the `@internal`). However, there has to be a way to inject the custom argument resolver manager passed to the `HttpKernel` into this class. As the controller resolver is instantiated outside the HttpKernel, the only way to inject it is using a setter.

In 3.0, all argument resolving logic should be removed from the controller resolver.